dwave/preprocessing/src/presolveimpl.hpp

```diff
@@ -928,6 +851,123 @@ class PresolverImpl {
         std::vector<Transform> transforms_;
     };
 
+    template <dimod::Sense Sense>
+    bool technique_domain_propagation(const constraint_type& constraint) {
+        // Dev note: if-branche(s) below rely on this static assert
+        static_assert(Sense == dimod::Sense::GE || Sense == dimod::Sense::LE,
+                      "Sense must be <= or >=; equality constraints should call both");
+
+        assert(constraint.sense() == dimod::Sense::EQ || constraint.sense() == Sense);
+
+        // todo: extend to quadratic
+        if (!constraint.is_linear()) {
+            return false;  // no changes
+        }
+
+        // Cannot use soft constraints to strengthen bounds.
+        if (constraint.is_soft()) {
+            return false;  // no changes
+        }
+
+        // In order to not fall into zeno's paradox here, we don't shrink bounds when
+        // the reduction is small.
+        static constexpr double BOUND_CHANGE_MINIMUM = 1.0e-3 * FEASIBILITY_TOLERANCE;
+
+        // Very large activities will create numeric issues. So we don't do domain
+        // propagation on those
+        static constexpr double MAX_ACTIVITY = 1.0e10;
```

Comment:
Could you tell more about the reason behind choosing 1.0e10 as MAX_ACTIVITY? Why not a larger number like 1.0e30 which is INF